include/aspect/particle/output/interface.h

```diff
@@ -86,10 +86,37 @@ namespace aspect
                                const double current_time) = 0;
 
           /**
+           * Returns the particle file name taking into account the file index as well as the mpi proccess.
+           */
+          virtual
+          const std::string
+          get_file_name ();
```

Comment:
The default implementation is   
``` 
     template <int dim>
      std::string
      Interface<dim>::get_file_name ()
      {
        return "particles-"
               + get_file_index()
               + "."
               + Utilities::int_to_string(Utilities::MPI::this_mpi_process(this->get_mpi_communicator()), 4)
               + output_file_suffix;
      }
```
 
Right, if the return type is not a pointer, then it shouldn;t be of type const but if the function itself doesn;t modify any provate or protected variables, then the function is of type const.